lib/codecs/tests/gelf.rs

```diff
@@ -0,0 +1,182 @@
+use bytes::Bytes;
+use chrono::{DateTime, NaiveDateTime, Utc};
+use codecs::{
+    decoding::format::gelf_fields::*, decoding::format::Deserializer, GelfDeserializerConfig,
+};
+use lookup::path;
+use pretty_assertions::assert_eq;
+use serde_json::json;
+use smallvec::SmallVec;
+use value::Value;
+use vector_core::{config::log_schema, event::Event};
+
+fn deserialize_gelf_input(input: &serde_json::Value) -> vector_core::Result<SmallVec<[Event; 1]>> {
+    let config = GelfDeserializerConfig;
+    let deserializer = config.build();
+    let buffer = Bytes::from(serde_json::to_vec(&input).unwrap());
+    deserializer.parse(buffer)
+}
+
+/// Validates all the spec'd fields of GELF are deserialized correctly.
+#[test]
```

Comment:
Should we look to move the `native` tests as well (in another PR) @pablosichert

Reply:
I think @lukesteensen put the `native` test under the crate level `tests` folder since those are bit more integrated rather than just plain unit tests. The distinction is not very clear cut, however 😄